### PR TITLE
Test fixture API for end to end tests

### PR DESCRIPTION
**Pull request by elcct**
### Description of change

A new set of endpoints has been added to aid end to end testing. These endpoints are enabled when
`ALLOW_TEST_FIXTURE_SETUP` environment variable is set. These endpoints are not authenticated and
should not be used in an environment other than tests.

`POST /testfixtureapi/reset-fixtures/`: Removes all database objects except Metadata and resets to
the state in the `test_data.yaml` fixtures file.

`POST /testfixtureapi/create-user/`: Creates a new user and authenticates provided token. It expects payload
in the following format:

```
{
        "first_name": <first name>,
        "last_name": <last name>,
        "email": <email>,
        "dit_team_id: <DIT team id>,
        "sso_email_user_id": <sso email user id>,
        "token": <desired token>
}
```

`POST /testfixtureapi/load-fixture/`: Loads database fixture. It expects fixture payload in the following format:
```
{
    "fixture": [
        {
            "model": "company.advisor",
            "pk": "413a608e-84a4-11e6-ea22-56b6b6499622",
            "fields":
            {
                "email": "abc@def",
                "first_name": "First name",
                "last_name": "Last name",
                "dit_team": "162a3959-9798-e211-a939-e4115bead28a"
            }
        }
    ]
}
```

### Checklist

* [x] If this is a releasable change, has a news fragment been added?

  <details>
  <summary>Explanation</summary>
  
  A news fragment is required for any releasable change (i.e. code that runs in or affects production) so that a corresponding changelog entry is added when releasing.
  
  Check [changelog/README.md](https://github.com/uktrade/data-hub-api/blob/master/changelog/README.md) for instructions.
  
  </details>
  
* [x] Has this branch been rebased on top of the current `develop` branch?

  <details>
  <summary>Explanation</summary>
  
  The branch should not be stale or have conflicts at the time reviews are requested.
  
  </details>

* [x] Is the CircleCI build passing?

### General points

<details>
<summary><strong>Other things to check</strong></summary><p></p>

* Make sure `fixtures/test_data.yaml` is maintained when updating models
* Consider the admin site when making changes to models
* Use select-/prefetch-related field lists in views and search apps, and update them when fields are added
* Make sure the README is updated e.g. when adding new environment variables

</details>

See [docs/CONTRIBUTING.md](https://github.com/uktrade/data-hub-api/blob/develop/docs/CONTRIBUTING.md) for more guidelines.

### CODE REVIEW

**Review by mcbhenwood — APPROVED** [review]
Very nice

File: changelog/e2e-endpoints.api.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+A new set of endpoints has been added to aid end to end testing. These endpoints are enabled when
+`ALLOW_TEST_FIXTURE_SETUP` environment variable is set. These endpoints are not authenticated and
+should not be used in an environment other than tests.
+
+`POST /testfixtureapi/reset-fitxtures/`: Removes all database objects except Metadata and resets to
```
Comment (rachaelcodes) [inline changelog/e2e-endpoints.api.md:5]: oops - a typo here `reset-fixtures` (I think it's ok in the rest of the PR)

File: changelog/e2e-endpoints.api.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+the state in the `test_data.yaml` fixtures file.
+
+`POST /testfixtureapi/create_user/`: Creates a new user and authenticates provided token. It expects payload
```
Comment (rachaelcodes) [inline changelog/e2e-endpoints.api.md:8]: I think the path here is `.../create-user/` ?

**Review by ambrozic — APPROVED** [review]

File: changelog/e2e-endpoints.api.md (hunk 1 above)
Comment (elcct) [reply changelog/e2e-endpoints.api.md]: Nice! thank you

File: changelog/e2e-endpoints.api.md (hunk 2 above)
Comment (elcct) [reply changelog/e2e-endpoints.api.md]: Yes! thank you

**Pull request merged**